common/task/weightedRoundRobinTaskScheduler.go

```diff
@@ -252,3 +265,43 @@ func (w *weightedRoundRobinTaskSchedulerImpl) notifyDispatcher() {
 		// do not block if there's already a notification
 	}
 }
+
+func (w *weightedRoundRobinTaskSchedulerImpl) getWeights() map[int]int {
+	return w.weights.Load().(map[int]int)
+}
+
+func (w *weightedRoundRobinTaskSchedulerImpl) updateWeights() {
+	ticker := time.NewTicker(defaultUpdateWeightsInterval)
+	for {
+		select {
+		case <-ticker.C:
+			weights, err := convertWeightsFromDynamicConfig(w.options.Weights())
+			if err != nil {
+				w.logger.Error("failed to update weight for round robin task scheduler", tag.Error(err))
+			} else {
+				w.weights.Store(weights)
+			}
+		case <-w.shutdownCh:
+			ticker.Stop()
+			return
+		}
+	}
+}
+
+func convertWeightsFromDynamicConfig(
```

Comment:
It looks like a misconfig will cause task processing errors. 
Shall we eat the convertConfig error, and only log the error and return default config?

Reply:
If the weight is invalid during server start up, then the start up logic will fail and server will panic.
If server has already been started and the weight is updated to an invalid value, then then update weight logic in the scheduler will be a no-op (keep using the existing weight) and print an error log. 
Does the above behavior make sense to you? Thoughts?

Reply:
ok, make sense.